alicloud/resource_alicloud_cs_managed_kubernetes.go

```diff
@@ -0,0 +1,657 @@
+			},
+			"worker_auto_renew": &schema.Schema{
+				Type:             schema.TypeBool,
+				Default:          false,
+				Optional:         true,
+				DiffSuppressFunc: csKubernetesWorkerPostPaidDiffSuppressFunc,
+			},
+			"worker_auto_renew_period": &schema.Schema{
+				Type:             schema.TypeInt,
+				Optional:         true,
+				Default:          1,
+				ValidateFunc:     validateAllowedIntValue([]int{1, 2, 3, 6, 12}),
+				DiffSuppressFunc: csKubernetesWorkerPostPaidDiffSuppressFunc,
+			},
+			"install_cloud_monitor": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"kube_config": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"client_cert": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"client_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"cluster_ca_cert": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			// 'version' is a reserved parameter and it just is used to test. No Recommendation to expose it.
+			"version": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"worker_nodes": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"private_ip": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"security_group_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vpc_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudCSManagedKubernetesCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	invoker := NewInvoker()
+
+	args, err := buildManagedKubernetesArgs(d, meta)
+	if err != nil {
+		return err
+	}
+	if err := invoker.Run(func() error {
+		raw, err := client.WithCsClient(func(csClient *cs.Client) (interface{}, error) {
+			return csClient.CreateKubernetesCluster(common.Region(client.RegionId), args)
+		})
+		if err != nil {
+			return err
+		}
+		cluster, _ := raw.(cs.ClusterCreationResponse)
+		d.SetId(cluster.ClusterID)
+		return nil
+	}); err != nil {
+		return fmt.Errorf("Creating ManagedKubernetes Cluster got an error: %#v", err)
+	}
+
+	if err := invoker.Run(func() error {
+		_, err := client.WithCsClient(func(csClient *cs.Client) (interface{}, error) {
+			return nil, csClient.WaitForClusterAsyn(d.Id(), cs.Running, 3600)
+		})
+		return err
+	}); err != nil {
+		return fmt.Errorf("Waitting for ManagedKubernetes cluster %#v got an error: %#v", cs.Running, err)
+	}
+
+	return resourceAlicloudCSManagedKubernetesUpdate(d, meta)
```

Comment:
I think create method can return read method directly and the update method can remove '!d.IsNewResource()'